core/src/bmi.jl

```diff
@@ -54,9 +54,28 @@ function BMI.initialize(T::Type{Model}, config::Config)::Model
         config.solver.maxiters,
     )
 
+    set_initial_controlled_parameters(parameters.control, u0, integrator)
+
     return Model(integrator, config, saved_flow)
 end
 
+function set_initial_controlled_parameters(
```

Comment:
This seems a mutating function, might be good to add ! to the function name. If so, you might want to reorder the arguments too (mutating arguments first).